frameworks/rooch-nursery/sources/inscribe_factory.move

```diff
@@ -0,0 +1,927 @@
+        if (is_err(&is_valid_result)) {
+            return is_valid_result
+        };
+
+        let attributes = get_SFT_attributes(metadata);
+
+        let generator = get_SFT_string_attribute(&attributes, b"generator");
+        let factory = get_SFT_string_attribute(&attributes, b"factory");
+        if (option::is_none(&generator) && option::is_none(&factory)) {
+            simple_map::drop(attributes);
+            return err(b"metadata.attributes.generator or metadata.attributes.factory is required")
+        };
+
+        if (option::is_some(&generator) && option::is_some(&factory)) {
+            simple_map::drop(attributes);
+            return err(b"metadata.attributes.generator and metadata.attributes.factory can not exist at the same time")
+        };
+        if (option::is_some(&generator)) {
+            let is_valid_result= is_valid_generator_uri(option::borrow(&generator));
+            if (is_err(&is_valid_result)) {
+                simple_map::drop(attributes);
+                return is_valid_result
+            };
+        };
+        if (option::is_some(&factory)) {
+            //TODO check factory
+        };
+
+        simple_map::drop(attributes);
+        ok(true)
+    }
+
+    fun is_valid_generator_uri(generator_uri: &String) : Result<bool> {
+        let index = string::index_of(generator_uri, &std::string::utf8(b"/inscription/"));
+        if (index != 0) {
+            return err(b"metadata.attributes.generator not start with /inscription/")
+        };
+
+        let inscription_id_str = string::sub_string(generator_uri, vector::length(&b"/inscription/"), string::length(generator_uri));
+        let inscription_id_option = ord::parse_inscription_id(&inscription_id_str);
+        if (option::is_none(&inscription_id_option)) {
+            return err(b"metadata.attributes.generator inscription_id parse fail")
+        };
+
+        let inscription_id = option::extract(&mut inscription_id_option);
+        if (!ord::exists_inscription(inscription_id)) {
+            return err(b"metadata.attributes.generator inscription not exists")
+        };
+
+        if (!ord::exists_metaprotocol_validity(inscription_id)) {
+            return err(b"metadata.attributes.generator inscription metaprotocol validity not exists")
+        };
+
+        let metaprotocol_validity = ord::borrow_metaprotocol_validity(inscription_id);
+
+        let is_match = ord::metaprotocol_validity_protocol_match<Bitseed>(metaprotocol_validity);
+        if (!is_match) {
+            return err(b"metadata.attributes.generator inscription metaprotocol validity protocol not match")
+        };
+
+        let is_valid = ord::metaprotocol_validity_is_valid(metaprotocol_validity);
+        if (!is_valid) {
+            return err(b"metadata.attributes.generator inscription metaprotocol validity not valid")
+        };
+        ok(true)
+    }
+
+    fun deploy_tick(metadata: &SimpleMap<String,vector<u8>>): (bool, Option<String>){
```

Comment:
The return value here is not changed to Result<bool>

Reply:
I will refactor it in a new PR.